src/server/middleware/csrfProtectionMiddleware.ts

```diff
@@ -0,0 +1,44 @@
+import * as url from "url";
+
+/**
+ * CSRF protection - ensure that state changing requests originate from us (our origin) or, as a
+ * fallback, that the referer's host matches. This is based on the recommendations here:
+ * https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet
+ * 
+ * CSRF protection should be applied to every transaction, which is why it is defined 
+ * as middleware, and should be applied for any route on the app. 
```

Comment:
nit: `applied to every transaction` and `applied for any route on the app` seem like they cover the same territory.

```suggestion
* CSRF protection verifies that the request originates from a place we expect.
* Defining it as middleware that's applied before every request ensures
* the check fires off before we route anywhere in the application.
```